File: T/src/main/scala/tinyscalautils/test/tagobjects/Fast.scala
package tinyscalautils.test.tagobjects

import org.scalatest.Tag

/**
  * A "not slow" tag.
  * This is so tests can be conditionally annotated with `Slow`.
  *
  * @see [[org.scalatest.tags.Slow]]
  * @since 1.5.0
  */
object Fast extends Tag("tinyscalautils.test.tags.Fast")

File: T/src/main/java/tinyscalautils/test/tags/Fast.java
package tinyscalautils.test.tags;

import org.scalatest.TagAnnotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * The annotation side of the `Fast` tag.
 *
 * @see tinyscalautils.test.tagobjects.Fast
 */
@TagAnnotation
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
public @interface Fast {}
